// @mastra/core/dist/auth/ee/index.cjs
'use strict';

var chunk2K2OYMQR_cjs = require('../../chunk-2K2OYMQR.cjs');
var chunkD6FHKJF5_cjs = require('../../chunk-D6FHKJF5.cjs');



Object.defineProperty(exports, "ACTIONS", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.ACTIONS; }
});
Object.defineProperty(exports, "DEFAULT_ROLES", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.DEFAULT_ROLES; }
});
Object.defineProperty(exports, "MastraFGAPermissions", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.MastraFGAPermissions; }
});
Object.defineProperty(exports, "PERMISSIONS", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.PERMISSIONS; }
});
Object.defineProperty(exports, "PERMISSION_PATTERNS", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.PERMISSION_PATTERNS; }
});
Object.defineProperty(exports, "RESOURCES", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.RESOURCES; }
});
Object.defineProperty(exports, "StaticRBACProvider", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.StaticRBACProvider; }
});
Object.defineProperty(exports, "buildCapabilities", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.buildCapabilities; }
});
Object.defineProperty(exports, "getDefaultRole", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.getDefaultRole; }
});
Object.defineProperty(exports, "hasPermission", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.hasPermission; }
});
Object.defineProperty(exports, "isAuthenticated", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.isAuthenticated; }
});
Object.defineProperty(exports, "isDevEnvironment", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.isDevEnvironment; }
});
Object.defineProperty(exports, "isEEEnabled", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.isEEEnabled; }
});
Object.defineProperty(exports, "isEELicenseValid", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.isEELicenseValid; }
});
Object.defineProperty(exports, "isFeatureEnabled", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.isFeatureEnabled; }
});
Object.defineProperty(exports, "isLicenseValid", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.isLicenseValid; }
});
Object.defineProperty(exports, "isValidPermissionPattern", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.isValidPermissionPattern; }
});
Object.defineProperty(exports, "matchesPermission", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.matchesPermission; }
});
Object.defineProperty(exports, "resolvePermissions", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.resolvePermissions; }
});
Object.defineProperty(exports, "resolvePermissionsFromMapping", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.resolvePermissionsFromMapping; }
});
Object.defineProperty(exports, "validateLicense", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.validateLicense; }
});
Object.defineProperty(exports, "validatePermissions", {
  enumerable: true,
  get: function () { return chunk2K2OYMQR_cjs.validatePermissions; }
});
Object.defineProperty(exports, "FGADeniedError", {
  enumerable: true,
  get: function () { return chunkD6FHKJF5_cjs.FGADeniedError; }
});
Object.defineProperty(exports, "checkFGA", {
  enumerable: true,
  get: function () { return chunkD6FHKJF5_cjs.checkFGA; }
});
