// @mastra/core/dist/utils.cjs
'use strict';

var chunkAX6D722J_cjs = require('./chunk-AX6D722J.cjs');
var chunkPAMAGEYY_cjs = require('./chunk-PAMAGEYY.cjs');
var chunkRLPRSIR2_cjs = require('./chunk-RLPRSIR2.cjs');



Object.defineProperty(exports, "checkEvalStorageFields", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.checkEvalStorageFields; }
});
Object.defineProperty(exports, "createMastraProxy", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.createMastraProxy; }
});
Object.defineProperty(exports, "deepEqual", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.deepEqual; }
});
Object.defineProperty(exports, "deepMerge", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.deepMerge; }
});
Object.defineProperty(exports, "delay", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.delay; }
});
Object.defineProperty(exports, "ensureSerializable", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.ensureSerializable; }
});
Object.defineProperty(exports, "ensureToolProperties", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.ensureToolProperties; }
});
Object.defineProperty(exports, "generateEmptyFromSchema", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.generateEmptyFromSchema; }
});
Object.defineProperty(exports, "getNestedValue", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.getNestedValue; }
});
Object.defineProperty(exports, "isCoreMessage", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.isCoreMessage; }
});
Object.defineProperty(exports, "isUiMessage", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.isUiMessage; }
});
Object.defineProperty(exports, "isZodType", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.isZodType; }
});
Object.defineProperty(exports, "makeCoreTool", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.makeCoreTool; }
});
Object.defineProperty(exports, "makeCoreToolV5", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.makeCoreToolV5; }
});
Object.defineProperty(exports, "maskStreamTags", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.maskStreamTags; }
});
Object.defineProperty(exports, "omitKeys", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.omitKeys; }
});
Object.defineProperty(exports, "parseFieldKey", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.parseFieldKey; }
});
Object.defineProperty(exports, "parseSqlIdentifier", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.parseSqlIdentifier; }
});
Object.defineProperty(exports, "removeUndefinedValues", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.removeUndefinedValues; }
});
Object.defineProperty(exports, "resolveSerializedZodOutput", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.resolveSerializedZodOutput; }
});
Object.defineProperty(exports, "safeStringify", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.safeStringify; }
});
Object.defineProperty(exports, "selectFields", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.selectFields; }
});
Object.defineProperty(exports, "setNestedValue", {
  enumerable: true,
  get: function () { return chunkAX6D722J_cjs.setNestedValue; }
});
Object.defineProperty(exports, "fetchWithRetry", {
  enumerable: true,
  get: function () { return chunkPAMAGEYY_cjs.fetchWithRetry; }
});
Object.defineProperty(exports, "getZodDef", {
  enumerable: true,
  get: function () { return chunkRLPRSIR2_cjs.getZodDef; }
});
Object.defineProperty(exports, "getZodTypeName", {
  enumerable: true,
  get: function () { return chunkRLPRSIR2_cjs.getZodTypeName; }
});
Object.defineProperty(exports, "isZodArray", {
  enumerable: true,
  get: function () { return chunkRLPRSIR2_cjs.isZodArray; }
});
Object.defineProperty(exports, "isZodObject", {
  enumerable: true,
  get: function () { return chunkRLPRSIR2_cjs.isZodObject; }
});
