// @mastra/core/dist/tools/index.cjs
'use strict';

var chunkKJBMTK5B_cjs = require('../chunk-KJBMTK5B.cjs');
var chunkM4KUR53T_cjs = require('../chunk-M4KUR53T.cjs');



Object.defineProperty(exports, "ToolStream", {
  enumerable: true,
  get: function () { return chunkKJBMTK5B_cjs.ToolStream; }
});
Object.defineProperty(exports, "MASTRA_TOOL_MARKER", {
  enumerable: true,
  get: function () { return chunkM4KUR53T_cjs.MASTRA_TOOL_MARKER; }
});
Object.defineProperty(exports, "Tool", {
  enumerable: true,
  get: function () { return chunkM4KUR53T_cjs.Tool; }
});
Object.defineProperty(exports, "createTool", {
  enumerable: true,
  get: function () { return chunkM4KUR53T_cjs.createTool; }
});
Object.defineProperty(exports, "isProviderDefinedTool", {
  enumerable: true,
  get: function () { return chunkM4KUR53T_cjs.isProviderDefinedTool; }
});
Object.defineProperty(exports, "isProviderTool", {
  enumerable: true,
  get: function () { return chunkM4KUR53T_cjs.isProviderTool; }
});
Object.defineProperty(exports, "isValidationError", {
  enumerable: true,
  get: function () { return chunkM4KUR53T_cjs.isValidationError; }
});
Object.defineProperty(exports, "isVercelTool", {
  enumerable: true,
  get: function () { return chunkM4KUR53T_cjs.isVercelTool; }
});
