// @mastra/core/dist/background-tasks/index.cjs
'use strict';

var chunkEBDGUBJJ_cjs = require('../chunk-EBDGUBJJ.cjs');



Object.defineProperty(exports, "BackgroundTaskManager", {
  enumerable: true,
  get: function () { return chunkEBDGUBJJ_cjs.BackgroundTaskManager; }
});
Object.defineProperty(exports, "backgroundOverrideJsonSchema", {
  enumerable: true,
  get: function () { return chunkEBDGUBJJ_cjs.backgroundOverrideJsonSchema; }
});
Object.defineProperty(exports, "backgroundOverrideZodSchema", {
  enumerable: true,
  get: function () { return chunkEBDGUBJJ_cjs.backgroundOverrideZodSchema; }
});
Object.defineProperty(exports, "createBackgroundTask", {
  enumerable: true,
  get: function () { return chunkEBDGUBJJ_cjs.createBackgroundTask; }
});
Object.defineProperty(exports, "generateBackgroundTaskSystemPrompt", {
  enumerable: true,
  get: function () { return chunkEBDGUBJJ_cjs.generateBackgroundTaskSystemPrompt; }
});
Object.defineProperty(exports, "resolveBackgroundConfig", {
  enumerable: true,
  get: function () { return chunkEBDGUBJJ_cjs.resolveBackgroundConfig; }
});
